### PR TITLE
feat(FWS-90): Ability to cancel or reschedule appointment

### PR DESCRIPTION
**Pull request by emilegirardGit**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-90

Context:
This ticket is about user being able to cancel and reschedule appointments

Changes
- Added cancel button for requested appointments
- Added cancel button for scheduled appointments
- Added reschedule button for schedule appointments
- Added reschedule and canceling functions

UI Changes
cancel
Reschedule

### CODE REVIEW

File: fitwsarah_frontend/src/components/clientProfile/Appointment.js [hunk 1]
```diff
@@ -31,14 +49,75 @@ function Appointment( {appointment, accessToken} ) {
...
+
+    const date_diff_indays = function (date1, date2) {
+        const dt1 = new Date(date1);
+        const dt2 = new Date(date2);
+        return Math.floor(
+            (Date.UTC(dt2.getFullYear(), dt2.getMonth(), dt2.getDate()) -
+                Date.UTC(dt1.getFullYear(), dt1.getMonth(), dt1.getDate())) /
+                    (1000 * 60 * 60 * 24)
+        );
+    };
```
Comment (Eli293) [inline fitwsarah_frontend/src/components/clientProfile/Appointment.js:88]: What does this function do?

File: fitwsarah_frontend/src/views/ProfilePage/Account.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ function Profile() {
         if (accessToken) {
             getAppointmentsByUserId(extractAfterPipe(user.sub), status);
         }
-    }, [user, status]);
+    }, [status, accessToken]);
```
Comment (Eli293) [inline fitwsarah_frontend/src/views/ProfilePage/Account.js:61]: Is user important here?

**Review by Eli293 — APPROVED** [review]
Looks good.

File: fitwsarah_frontend/src/components/clientProfile/Appointment.js (hunk 1 above)
Comment (emilegirardGit) [reply fitwsarah_frontend/src/components/clientProfile/Appointment.js:88]: It calculate the number of days between the appointment date and todays date

File: fitwsarah_frontend/src/views/ProfilePage/Account.js (hunk 2 above)
Comment (emilegirardGit) [reply fitwsarah_frontend/src/views/ProfilePage/Account.js:61]: No because if the accessToken is set the user will be set as well

**Review by Abdur-Siddiqui — APPROVED** [review]
Good job

**Review by Abdur-Siddiqui — APPROVED** [review]
Good job

**Pull request merged**